src/pushsource/_impl/source.py

```diff
@@ -29,14 +31,42 @@ class SourceUrlError(ValueError):
 
 
 class SourceWrapper(object):
+    timeout = 900
```

Comment:
The values are now set by env variables. Though I moved them directly to __iter__ method so that env varialbes set by tests would be read correctly. I also disabled the feature by default.

Anyway, I have a hard time imagining what "In our case it means we set an env var in pub target settings to explicitly enable the timeout we want." will look like. Aren't source and target separate entities? Can target even influence what happens in source code? And do target settings even have the ability to set env variables? Is this some new logic that will have to be implemented somewhere?

Reply:
A target is just a bunch of metadata.  Many Pub tasks use some of that metadata to adjust arguments or environment variables at the beginning of the task.

Target settings have the ability to set env vars, it was implemented a while ago in RHELDST-7231.  You were reviewer on that, so if you want to improve awareness of what's happening in the code you could consider putting more effort on keeping up with review invites.